pkg/k8sutils/pod.go

```diff
@@ -3,43 +3,125 @@ package k8sutils
 import (
 	"bytes"
 	"context"
+	"fmt"
 	"io"
 	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"strings"
 )
 
 // GetLogsOfPod returns the k8s logs of a job in a namespace
 func (k8s *k8sImpl) GetLogsOfPod(jobName string, namespace string) (string, error) {
 
-	// TODO include the logs of the initcontainer
-
-	podLogOpts := v1.PodLogOptions{}
-
 	list, err := k8s.clientset.CoreV1().Pods(namespace).List(context.TODO(), metav1.ListOptions{
 		LabelSelector: "job-name=" + jobName,
 	})
 	if err != nil {
 		return "", err
 	}
 
-	logs := ""
+	var logs strings.Builder
 
 	for _, pod := range list.Items {
 
-		req := k8s.clientset.CoreV1().Pods(namespace).GetLogs(pod.Name, &podLogOpts)
-		podLogs, err := req.Stream(context.TODO())
-		if err != nil {
-			return "", err
+		// Query all logs from containers that have terminated and therefore already had the chance to
+		// produce logs, otherwise the k8s api will return an error
+		for _, container := range getTerminatedContainersWithStatusOfPod(pod) {
+
+			// Query logs of the current selected container
+			logsOfContainer, err := getLogsOfContainer(k8s, pod, namespace, container.name)
+			if err != nil {
+				return "", err
+			}
```

Comment:
Returning here means we might lose existing `logs` of previous (terminated) containers. I'd rather add a message to `logs` with the error message contained in `err`.